src/main/java/com/google/starfish/services/FavoriteNoteService.java

```diff
@@ -0,0 +1,152 @@
+    if (userId == null) return;
+    try (Connection conn = pool.getConnection()) {
+      try {
+        conn.setAutoCommit(false);
+        String stmt =
+            "INSERT INTO favorite_notes ( "
+                + "note_id,"
+                + "user_id ) "
+          + "VALUES ( "
+                + "?,"
+                + "? ); ";
+        try (PreparedStatement insertStmt = conn.prepareStatement(stmt)) {
+          insertStmt.setLong(1, noteId);
+          insertStmt.setString(2, userId);
+          System.out.println(insertStmt);
+          insertStmt.execute();
+          conn.commit();
+        }
+      } catch(SQLException ex) {
+        if (conn != null) {
+          try {
+            System.err.print("Transaction is being rolled back.");
+            conn.rollback();
+          } catch (SQLException excep) {
+            System.err.print(excep);
+          }
+        }
+      }
+    } catch (SQLException ex) {
+      System.err.print(ex);
+    }
+  }
+
+  /** Gets an array of note objects that have been favorited by passed user id */
+  public Note[] getFavoriteNotesByUserId(DataSource pool, String userId) throws SQLException {
+    List<Note> notes = new ArrayList<>();
+    try (Connection conn = pool.getConnection()) {
+      String stmt = 
+          "SELECT  a.* "
+        + "FROM "
+          + Table.NOTES.getSqlTable() + " AS a "
+          + "INNER JOIN (SELECT * "
+                      + "FROM " + Table.FAVORITE_NOTES.getSqlTable() + " "
+                      + "WHERE user_id=?) "
+          + "as b ON a.id=b.note_id;";
+      try (PreparedStatement favNotesStmt = conn.prepareStatement(stmt)) {
+        favNotesStmt.setString(1, userId);
+        ResultSet rs = favNotesStmt.executeQuery();
+        while (rs.next()) {
+          long noteId = rs.getLong("id");
+          String authorId = rs.getString("author_id");
+          String school = rs.getString("school");
+          String course = rs.getString("course");
+          String title = rs.getString("title");
+          String sourceUrl = rs.getString("title");
+          String pdfSource = rs.getString("pdf_source");
+          Date dateCreated = rs.getDate("date_created");
+          long numDownloads = rs.getLong("num_downloads");
+          long numFavorites = getNumFavoritesByNoteId(pool, noteId);
+
+          Note thisNote = new Note.Builder()
+                              .setId(noteId)
+                              .setAuthorId(authorId)
+                              .setRequiredLabels(school, course)
+                              .setNoteTitle(title)
+                              .setSourceUrl(sourceUrl)
+                              .setOptionalPdfSource(pdfSource)
+                              .setDateCreated(dateCreated)
+                              .setNumDownloads(numDownloads)
+                              .setNumFavorites(numFavorites)
+                              .build();
+          notes.add(thisNote);
```

Comment:
I noticed that this logic is also used in the NotesService class - do you think it'd be beneficial to make a construct note method or something that would take in the ResultSet and return the note?

Reply:
Yes potentially, I think it could be a method in the NoteService class and then it's used as a dependency in the FavoriteNoteService class. I'll make a change.